angularjs-portal-frame/src/main/webapp/portal/main/controllers.js

```diff
@@ -61,6 +62,19 @@ define(['angular'], function(angular) {
       that.user = result;
     });
   }]);
+  
+  app.controller('WelcomeController', ['$localStorage', '$sessionStorage','$scope', '$document', 'NAMES', '$modal', function($localStorage, $sessionStorage, $scope, $document, NAMES, $modal) {
+    $scope.openModal = function() {
+      if ($localStorage.hasSeenWelcome === false) {
+        $modal.open({
+          animation: $scope.animationsEnabled,
+          templateUrl: require.toUrl('../../my-app/layout/partials/welcome.html'),
```

Comment:
Hmmm, this breaks frame apps if the child app didn't have this file. Why not have the partial in frame?